hazelcast/src/main/java/com/hazelcast/internal/eviction/ExpirationManager.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.eviction;
+
+import com.hazelcast.cluster.ClusterState;
+import com.hazelcast.core.LifecycleEvent;
+import com.hazelcast.core.LifecycleListener;
+import com.hazelcast.spi.NodeEngine;
+import com.hazelcast.spi.TaskScheduler;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static com.hazelcast.util.Preconditions.checkPositive;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+/**
+ * This task class is responsible for gradual cleanup of expired entries. For this purpose it uses a background task.
+ * @since 3.3
```

Comment:
Maybe we can add the info, that it's used since 3.3 for IMap and since 3.11 for ICache. Information like this gets lost easily and sooner or later someone will ask for it :)